System/Desk/DIWorkSpaceDock.ts

```diff
@@ -118,27 +118,19 @@ export class DIWorkSpaceDock extends DIView {
 
     async update() {
         this.unplugChildViews();
-        for (let i = 0; i < secretaryInstance.workSpaces.length; i++) {
-            this.addChildView(secretaryInstance.workSpaces[i].icon);
-            secretaryInstance.workSpaces[i].icon.y = i * 64;
-            if (secretaryInstance.workSpaces[i].icon.events.length < 1) {
-                // TODO: use fat arrow function instead of .bind
-                secretaryInstance.workSpaces[i].icon.events.push(
-                    // eslint-disable-next-line @typescript-eslint/no-loop-func
-                    new DeskEvent(secretaryInstance.workSpaces[i].icon.body, "click", () => {
-                        // @ts-ignore
-                        secretaryInstance.workSpaces[i].desk.workSpaceDock.clicked(this);
-                    })
-                );
+        secretaryInstance.workSpaces.forEach((_workSpace: WorkSpace, idx: number) => {
```

Comment:
I think we can replace this `.forEach` loop with 

`for (const [workSpace, idx] of secretaryInstance.workSpaces.entries())`

What are your thoughts @qlqhwb?

Reply:
same as above